File: packages/python/examples/16_llm_anthropic.py
"""
Example 16: Anthropic-Powered Agent

Demonstrates using the anthropic_skill decorator for Claude-powered skills.

Prerequisites:
    pip install a2a-lite[anthropic]
    export ANTHROPIC_API_KEY=your-key-here

Run:
    python examples/16_llm_anthropic.py
"""
from a2a_lite import Agent
from a2a_lite.llm import anthropic_skill

agent = Agent(
    name="Claude Agent",
    description="An agent powered by Anthropic's Claude",
    version="1.0.0",
)


@agent.skill("analyze", description="Analyze text with Claude")
@anthropic_skill(
    model="claude-sonnet-4-5-20250929",
    system_prompt="You are an expert analyst. Provide clear, structured analysis.",
    max_tokens=1024,
)
async def analyze(text: str) -> str:
    """Analyze the provided text using Claude."""
    ...  # Handled by the anthropic_skill decorator


@agent.skill("code_review", description="Review code with Claude")
@anthropic_skill(
    model="claude-sonnet-4-5-20250929",
    system_prompt=(
        "You are an expert code reviewer. Review the code for bugs, "
        "security issues, and style. Be concise."
    ),
    max_tokens=2048,
    temperature=0.3,
)
async def code_review(text: str) -> str:
    """Review code and provide feedback."""
    ...  # Handled by the anthropic_skill decorator


@agent.skill("chat", description="Chat with Claude (streaming)", streaming=True)
@anthropic_skill(
    model="claude-sonnet-4-5-20250929",
    system_prompt="You are a helpful, friendly assistant.",
    streaming=True,
)
async def chat(message: str) -> str:
    """Stream a conversation with Claude."""
    ...  # Handled by the anthropic_skill decorator


if __name__ == "__main__":
    agent.run(port=8787)

File: packages/typescript/examples/16_llm_anthropic.ts
/**
 * LLM integration with Anthropic.
 *
 * Uses the `anthropicSkill` factory to create skill handlers that call the
 * Anthropic messages API. No manual HTTP calls needed.
 *
 * Prerequisites:
 *   npm install @anthropic-ai/sdk
 *   export ANTHROPIC_API_KEY="sk-ant-..."
 *
 * Run: npx ts-node examples/16_llm_anthropic.ts
 */
import { Agent, anthropicSkill } from '../src';

const agent = new Agent({
  name: 'AnthropicAgent',
  description: 'An agent powered by Anthropic Claude models',
});

// Analysis skill — good for structured reasoning tasks
agent.skill('analyze', anthropicSkill({
  model: 'claude-opus-4-6',
  systemPrompt: 'You are an expert analyst. Break down the given topic into key insights, implications, and actionable recommendations.',
  maxTokens: 1024,
}));

// Code review skill — lower temperature for precise feedback
agent.skill('code_review', anthropicSkill({
  model: 'claude-opus-4-6',
  systemPrompt: 'You are a senior software engineer conducting a code review. Identify bugs, suggest improvements, and highlight good patterns. Be specific and constructive.',
  temperature: 0.3,
  maxTokens: 2048,
}));

// Streaming chat skill — yields tokens as they arrive
agent.skill('chat', { streaming: true }, anthropicSkill({
  model: 'claude-opus-4-6',
  systemPrompt: 'You are a helpful assistant.',
  streaming: true,
}));

agent.run({ port: 8787 });
